src/ploigos_step_runner/step_implementers/automated_governance/rekor.py

```diff
@@ -0,0 +1,277 @@
+                "signature": {
+                    "format": "pgp",
+                    "content": self.base64_encode(sig_file),
+                    "publicKey": {
+                        "content": self.base64_encode(public_key_path)
+                    }
+                },
+                "data": {
+                    "content": base64_encoded_extra_data,
+                    "hash": {
+                        "algorithm": "sha256",
+                        "value": hash
+                    }
+                },
+                "extraData": base64_encoded_extra_data
+            }
+        }
+        return rekor_entry;
+
+    def get_gpg_key(self, sig_file, extra_data_file):
+        """Runs the step implemented by this StepImplementer.
+
+        Returns
+        -------
+        StepResult
+            Object containing the dictionary results of this step.
+        """
+        gpg_stdout_result = StringIO()
+        gpg_stdout_callback = create_sh_redirect_to_multiple_streams_fn_callback([
+            sys.stdout,
+            gpg_stdout_result
+        ])
+        gpg_user = self.get_value('gpg-user')
+        sh.gpg(
+            '--armor',
+            '-u',
+            gpg_user,
+            '--output',
+            sig_file,
+            '--detach-sign',
+            extra_data_file,
+            _out=gpg_stdout_callback,
+            # _in=extra_data,
+            _err_to_out=True,
+            _tee='out'
+        )
+        return gpg_stdout_result
+
+    def upload_to_rekor(self, rekor_server, extra_data_file):
+        """Runs the step implemented by this StepImplementer.
+
+        Returns
+        -------
+        StepResult
+            Object containing the dictionary results of this step.
+        """
+        gpg_key = self.get_value('gpg-key')
+        rekor_entry = self.create_rekor_entry(gpg_key, extra_data_file)
+        print("Rekor Entry: " + str(rekor_entry))
+        # print("Rekor Entry Type: "+ str(type(rekor_entry)))
+        rekor_entry_path = Path(os.path.join(self.work_dir_path, 'entry.json'))
+
+        if rekor_entry_path.exists():
+            rekor_entry_path.unlink()
+        rekor_entry_path_name = os.path.join(self.work_dir_path, 'entry.json')
+        rekor_entry_path.write_text(json.dumps(rekor_entry))
+        rekor_upload_stdout_result = StringIO()
+        rekor_upload_stdout_callback = create_sh_redirect_to_multiple_streams_fn_callback([
+            sys.stdout,
+            rekor_upload_stdout_result
+        ])
+        rekor = sh.rekor(
+                'upload',
+                '--rekor_server',
+                rekor_server,
+                '--entry',
+                rekor_entry_path_name,
+                _out=rekor_upload_stdout_callback,
+                _err_to_out=True,
+                _tee='out'
+                )
+        rekor_uuid = str(rekor).split('/')[-1].strip(' \n')
+        return rekor_entry, rekor_uuid
+
+    def get_all_step_results_dict(self):
```

Comment:
shouldnt need this. since its just being dumped to json any ways get rid of this function and then use `self.workflow_result.write_results_to_json_file(..)`

Reply:
https://github.com/ploigos/ploigos-step-runner/pull/146/commits/b53a991b19f14c441622a5c8a6725e7bffd4579d